pkg/security/podsecuritypolicy/seccomp/mustrunas.go

```diff
@@ -0,0 +1,123 @@
+func NewWithSeccompProfile(allowedProfiles []string) (SeccompStrategy, error) {
+	return &withSeccompProfile{allowedProfiles}, nil
+}
+
+// Generate creates the profile based on policy rules.
+func (s *withSeccompProfile) Generate(pod *api.Pod) (string, error) {
+	// return the first non-wildcard profile
+	for _, p := range s.allowedProfiles {
+		if p != allowAnyProfile {
+			return p, nil
+		}
+	}
+	// if we reached this point then either there are no allowed profiles (empty slice)
+	// or the only thing in the slice is the wildcard.  In either case just return empty.
+	return "", nil
+}
+
+// ValidatePod ensures that the specified values on the pod fall within the range
+// of the strategy.
+func (s *withSeccompProfile) ValidatePod(pod *api.Pod) field.ErrorList {
+	allErrs := field.ErrorList{}
+	podSpecFieldPath := field.NewPath("pod", "spec", "securityContext", "seccompProfile")
+
+	if len(s.allowedProfiles) == 0 &&
+		pod.Spec.SecurityContext != nil &&
+		pod.Spec.SecurityContext.SeccompProfile != "" {
+
+		allErrs = append(allErrs, field.Forbidden(podSpecFieldPath, "seccomp may not be set"))
+		return allErrs
+	}
+
+	if pod.Spec.SecurityContext != nil {
+		if !isProfileAllowed(pod.Spec.SecurityContext.SeccompProfile, s.allowedProfiles) {
+			allErrs = append(allErrs, field.NotSupported(podSpecFieldPath.Child("securityContext", "seccompProfile"),
+				pod.Spec.SecurityContext.SeccompProfile, s.allowedProfiles))
+		}
+	}
+
+	return allErrs
+}
+
+// ValidateContainer ensures that the specified values on the container fall within
+// the range of the strategy.
+func (s *withSeccompProfile) ValidateContainer(pod *api.Pod, container *api.Container) field.ErrorList {
+	allErrs := field.ErrorList{}
+	fieldPath := field.NewPath("pod", "spec", "containers").Key(container.Name).Child("securityContext", "seccompProfile")
+
+	if len(s.allowedProfiles) == 0 &&
+		container.SecurityContext != nil &&
+		container.SecurityContext.SeccompProfile != "" {
+		allErrs = append(allErrs, field.Forbidden(fieldPath, "seccomp may not be set"))
+		return allErrs
+	}
+
+	if container.SecurityContext != nil {
+		if !isProfileAllowed(container.SecurityContext.SeccompProfile, s.allowedProfiles) {
+			allErrs = append(allErrs, field.NotSupported(fieldPath,
+				container.SecurityContext.SeccompProfile, s.allowedProfiles))
+		}
+	}
+
+	return allErrs
+}
+
+// getProfileForContainer returns the container's profile or, if it is not set, the pod's profile.
+func getProfileForContainer(container *api.Container, podProfile string) string {
+	if container.SecurityContext != nil && container.SecurityContext.SeccompProfile != "" {
+		return container.SecurityContext.SeccompProfile
+	}
+	return podProfile
+}
+
+// isProfileAllowed checks if profile is in allowedProfiles or if allowedProfiles
+// contains the wildcard.
+func isProfileAllowed(profile string, allowedProfiles []string) bool {
+	// not allowing any and nothing is set
+	if len(allowedProfiles) == 0 && profile == "" {
```

Comment:
"" should always be allowed. Or does the validation take place after the submission controller run?